### PR TITLE
Feat: Adding integration function for fleet

### PR DESCRIPTION
**Pull request by aagusuab**
### Description of your changes

Fixes #

I have:

- [ ] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [ ] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/controllers/apply_controller.go [hunk 1]
```diff
@@ -61,6 +61,17 @@ type ApplyWorkReconciler struct {
 	concurrency        int
 }
 
+func NewWorkController(hubClient client.Client, spokeDynamicClient dynamic.Interface, spokeClient client.Client, restMapper meta.RESTMapper, recorder record.EventRecorder, concurrency int) *ApplyWorkReconciler {
```
Comment (Arvindthiru) [inline pkg/controllers/apply_controller.go:64]: Can rename this to NewApplyWorkReconciler ?

File: pkg/controllers/manager.go [hunk 2]
```diff
@@ -69,10 +69,10 @@ func Start(ctx context.Context, hubCfg, spokeCfg *rest.Config, setupLog logr.Log
 		os.Exit(1)
 	}
 
-	if err = newWorkStatusReconciler(
+	if err = NewWorkStatusReconciler(
```
Comment (Arvindthiru) [inline pkg/controllers/manager.go:72]: You can use the constructors to create the reconcilers down below if you want

File: pkg/controllers/apply_controller.go (hunk 1 above)
Comment (aagusuab) [reply pkg/controllers/apply_controller.go]: Renamed.

File: pkg/controllers/manager.go (hunk 2 above)
Comment (aagusuab) [reply pkg/controllers/manager.go:72]: I have used the constructors to create the reconcilers.

**Review by Arvindthiru — APPROVED** [review]

**Pull request merged**